src/main/java/org/candlepin/model/ContentAccessCertificateCurator.java

```diff
@@ -129,4 +131,41 @@ private int deleteContentAccessCerts(Set<String> certIdsToDelete) {
         return removed;
     }
 
+
+    /**
+     * Lists all expired content access certificates that are not revoked.
+     *
+     * @return a list of expired certificates
+     */
+    @SuppressWarnings("unchecked")
+    public List<ExpiredCertificate> listAllExpired() {
```

Comment:
Something about using a custom projection object here isn't sitting well with me. I'm not sure we actually need it, and it seems like this (and a few other methods) are optimizations around a join or lookup later.

Reply:
Even after refactoring the query I don't see a good way to replace the projection without splitting it into two queries. Do you have any suggestion on how to do get rid of it?

Reply:
Just a note that we discussed this on IRC a few days ago, and the serials being exposed both to the REST API and Adapters API complicates this, and would be better done as a separate task

Reply:
Perhaps a naive question, but why do we need the custom object for this query? Does HQL not allow for selecting 2 columns from 2 joined tables? e.g. `SELECT c.id, s.id ...`?

Reply:
We don't. Using scalar projection would be fine. I just find DTO projections more descriptive compared to array you get from scalar projection.

Reply:
I suppose I can see both the pros/cons of this approach, but as someone reading this query for the first time, I was immediately trying to understand why is this needed, which is distracting and takes away from readability/descriptiveness (at least for me).
The fact we have a separate model class just for this is also not great. If this became a trend, then we'd have model classes mixed in with other model classes that aren't really entity models, but DTO projections. If we are to keep this DTO projection, I'd say we should put it under a sub-package like `model.projections` or something.

I'm fine with either way (keeping DTO projection, or using scalar projection), so we could keep it unless @Ceiu feels strongly about this

Reply:
It's not really a separate model. Just a view of model. It's useful when you only need read-only data and it's actually a recommended practice to use projections as they have performance benefits:
- You are not querying  as much data
- DTO and scalar projections are ~20% faster than entities (for the same amount of queried data) as Hibernate managed entities have huge overhead.

Reply:
Generally speaking, I don't like creating objects for the sake of creating objects. I initially called this area out because the logic throughout the chain is coupled down to the new expiry job, and the projection DTO seemed to be a result of that. However, in this case, I've since come to the conclusion that the logic isn't the problem here, but rather a result of our goofy model surrounding certs and serials.  In cases like this, where we have a clear need for a multi-column projection *and* we need the result of said projection outside of the curator, it's acceptable.

I'd still prefer we didn't have it overall, but the only performant way to remove it that I see involves fixing the cert model, which we've decided is a separate task.